test-case-reporting/test/e2e.test.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Knex from 'knex';
+import nock from 'nock';
+import request from 'supertest';
+
+import {Database, dbConnect} from '../src/db';
+import {app} from '../index';
+import {getFixtureAsString} from './fixtures';
+import {setupDb} from '../src/setup_db';
+
+jest.mock('../src/db', () => ({
+  dbConnect: () =>
+    Knex({
+      client: 'sqlite3',
+      connection: ':memory:',
+      useNullAsDefault: true,
+    }),
+}));
+
+describe('end-to-end', () => {
+  let db: Database;
+
+  beforeAll(async () => {
+    nock.disableNetConnect();
+    process.env = {
+      DB_UNIX_SOCKET: 'amp-test-cases/socket',
+      DB_USER: 'test_user',
+      DB_PASSWORD: 'test_password',
+      DB_NAME: 'test_results',
+      NODE_ENV: 'test',
+    };
+
+    db = dbConnect();
+    await setupDb(db);
+  });
+
+  afterAll(() => {
+    nock.enableNetConnect();
+    db.destroy();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    db('builds').truncate();
```

Comment:
Something to be aware of: unless you have `ON CASCADE DELETE` set for your foreign keys (you don't), trying to delete a row that is referenced by a foreign key on another table will fail. For instance, if you have a row in `builds` with `id=6`, and a row in `jobs` with `build_id=6`, trying to delete the build will fail. You'll have to delete entries in reverse-dependency order, so in this case, `jobs` have to be deleted before the `builds` they reference. You should be able to extrapolate for the remaining tables.

SQL has lots of little edge cases :P

Reply:
Nice explanation of `ON DELETE` actions, explained through a metaphor of businesses and the bourgeoisie:
https://dba.stackexchange.com/a/213239

Reply:
Done. I'll check out the link!